tests/datasets/auto_sales/mod.rs

```diff
@@ -0,0 +1,718 @@
+                col("year"),
+                col("manufacturer"),
+                round(vec![col("total_sales"), lit(4)]).alias("total_sales"),
+            ])? // Round the total sales to 4 decimal places.
+            .sort(vec![
+                col("year").sort(true, false),
+                col("total_sales").sort(false, false),
+            ])?; // Sort by year and descending total sales.
+
+        // Collect DataFrame results and transform them into a comparable format.
+        let expected_results: Vec<(i32, String, f64)> = df_result
+            .collect()
+            .await?
+            .into_iter()
+            .flat_map(|batch| {
+                let year_column = batch
+                    .column(0)
+                    .as_any()
+                    .downcast_ref::<Int32Array>()
+                    .unwrap();
+                let manufacturer_column = batch
+                    .column(1)
+                    .as_any()
+                    .downcast_ref::<StringArray>()
+                    .unwrap();
+                let total_sales_column = batch
+                    .column(2)
+                    .as_any()
+                    .downcast_ref::<Float64Array>()
+                    .unwrap();
+
+                (0..batch.num_rows())
+                    .map(move |i| {
+                        (
+                            year_column.value(i),
+                            manufacturer_column.value(i).to_owned(),
+                            total_sales_column.value(i),
+                        )
+                    })
+                    .collect::<Vec<_>>()
+            })
+            .collect();
+        let df_duration = df_start_time.elapsed();
+
+        // Print the results from both PostgreSQL and DataFrame for comparison.
+        print_utils::print_results(
+            vec![
+                "Year".to_string(),
+                "Manufacturer".to_string(),
+                "Total Sales".to_string(),
+            ],
+            "Pg_Analytics".to_string(),
+            &total_sales_results,
+            "DataFrame".to_string(),
+            &expected_results,
+        )
+        .await?;
+
+        // Compare the results with a small epsilon for floating-point precision.
+        let epsilon = 0.0001;
+        for ((pg_year, pg_manufacturer, pg_total), (df_year, df_manufacturer, df_total)) in
+            total_sales_results.iter().zip(expected_results.iter())
+        {
+            assert_eq!(pg_year, df_year, "Year mismatch");
+            assert_eq!(pg_manufacturer, df_manufacturer, "Manufacturer mismatch");
+            assert!(
+                (pg_total - df_total).abs() < epsilon,
+                "Total sales for year {} and manufacturer {} differ more than expected epsilon: PostgreSQL = {}, DataFrame = {}",
+                pg_year, pg_manufacturer, pg_total, df_total
+            );
+        }
+
+        tracing::info!("Time taken for Pg_Analytics query: {:?}", pg_duration);
```

Comment:
Tests should just be for asserting results. Performance times should be measured in separate benchmarks, let's remove this.

Reply:
Makes sense, I've removed the unnecessary parts.